d3b_dff_cli/modules/intake/request.py

```diff
@@ -0,0 +1,177 @@
+    }
+
+    field_id = default_ids[field]
+
+    meta_data = json.loads(meta_res.data)
+
+    for ticket_field in meta_data["fields"]:
+        if ticket_field["name"] == field:
+            for option in ticket_field["allowedValues"]:
+                if option["value"] == key:
+                    field_id = option["id"]
+
+    return field_id
+
+
+def intake_request(args):
+    """
+    Create data intake epic and return epic and transfer ticket ids.
+
+    Input:
+    - args (argparse.Namespace): Parsed command-line arguments
+
+    Output:
+    - epic_id (str): Epic ID
+    - transfer_id (str): Transfer ID
+    """
+
+    # jira internal id for the data intake epic
+    intake_epic_id = 10231
+
+    # jira internal id for AD project
+    project_id = 10147
+
+    epic_key, transfer_key = ["AD-0", "AD-0"]
+
+    http = urllib3.PoolManager()
+
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": f"Basic {args.auth}",
+    }
+
+    url = f"{args.jira_url}/rest/api/3/issue/"
+
+    # Required cutom fields:
+    study_field_key = "customfield_10136"
+    data_source_field_key = "customfield_10139"
+    program_field_key = "customfield_10140"
+
+    # translate between text and custom field allowed value id
+
+    # get issue type fields and required values
+    meta_url = f"{args.jira_url}/rest/api/3/issue/createmeta/{project_id}/issuetypes/{intake_epic_id}"
+    meta_res = http.request("GET", meta_url, headers=headers)
+    check_status(meta_res)
+
+    # convert from text to allowed value id
+    study_id = convert_keys_to_id(args.study, "Study", meta_res)
+    data_source_id = convert_keys_to_id(args.data_source, "Data Source", meta_res)
+    program_id = convert_keys_to_id(args.program, "Program", meta_res)
+
+    date = datetime.now()
+
+    # override default summary if args.summary provided
+    summary = None
+    if args.summary:
+        summary = args.summary
+    else:
+        summary = f"{args.study} data intake from {args.data_source} - {date}"
+
+    # add test label if not prd
+    if not args.prd:
+        summary = f"TEST {summary}"
+
+    # build post json
+    payload = json.dumps(
+        {
+            "fields": {
+                "project": {"id": project_id},
+                "issuetype": {"id": intake_epic_id},
+                "summary": summary,
+                study_field_key: [{"id": study_id}],
+                data_source_field_key: {"id": data_source_id},
+                program_field_key: {"id": program_id},
+            }
+        }
+    ).encode("utf-8")
+
+    # if args.post, post and make epic
+    if args.post:
+        response = http.request("POST", url, body=payload, headers=headers)
+
+        check_status(response)
+
+        # extract epic id and transfer id
+        epic_key = json.loads(response.data)["key"]
+
+        # get transfer key from epic key
+        epic_proj, epic_num = epic_key.split("-")
+        transfer_key = f"{epic_proj}-{int(epic_num)+1}"
```

Comment:
this might not work perfectly, especially if someone else is creating a ticket at the same time.  Its rare, but it could happen, since there's a slight delay between when the epic is created and when the automation runs to create the data transfer child. If someone else is creating a ticket at the same time, that might "take" the next key available.  That is to say, there's no real guarantee that just because the epic is AD-123, the transfer ticket will be AD-124.  Its likely, but not for sure.  

It would be more appropriate to do the wait/sleep, then check the epic to see if it has children and get the correct child based on ticket type, and get the transfer ticket ID from there.

Reply:
Agree with @calkinsh 's suggestion. Otherwise everything else looks good to me as well.